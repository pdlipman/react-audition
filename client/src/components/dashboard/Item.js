import React, { Component } from 'react';
import PropTypes from 'prop-types';

import Grid from 'material-ui/Grid';
import Typography from 'material-ui/Typography';
import blue from 'material-ui/colors/blue';

export class Item extends Component {
  static propTypes = {
    label: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.number,
    ]),
    style: PropTypes.object,
    className: PropTypes.string,
  };

  static defaultProps = {
    label: '',
    style: {},
    className: '',
  };

  render() {
    const {
      label,
      style,
      className,
    } = this.props;
    return (
      <Grid
        item
        xs={ 6 }
        style={ style }
        className={ className }
      >
        <Typography
          style={{ color: blue[500] }}
        >
          { label }
        </Typography>
      </Grid>
    );
  }
}

export default Item;
