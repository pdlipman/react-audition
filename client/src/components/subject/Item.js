import React, { Component } from 'react';

export default class Item extends Component {
  render () {
    return (
      <p>
        Item
      </p>
    );
  }
}

