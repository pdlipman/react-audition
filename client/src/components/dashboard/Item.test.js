import React from 'react';

import { shallow } from 'enzyme';
import { Item } from './Item';

describe('Item', () => {
  test('Item should render properly', () => {
    const wrapper = shallow(
      <Item />
    );
    expect(wrapper).toMatchSnapshot();
  });
});