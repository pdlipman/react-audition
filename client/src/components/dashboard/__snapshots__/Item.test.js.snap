// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item Item should render properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Item
    className=""
    label=""
    style={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <WithStyles(Typography)
        style={
                Object {
                        "color": "#2196f3",
                      }
        }
/>,
      "className": "",
      "item": true,
      "style": Object {},
      "xs": 6,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "",
        "style": Object {
          "color": "#2196f3",
        },
      },
      "ref": null,
      "rendered": "",
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <WithStyles(Typography)
          style={
                    Object {
                              "color": "#2196f3",
                            }
          }
/>,
        "className": "",
        "item": true,
        "style": Object {},
        "xs": 6,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "",
          "style": Object {
            "color": "#2196f3",
          },
        },
        "ref": null,
        "rendered": "",
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
